pycbc/inference/sampler/dynesty.py

```diff
@@ -175,10 +231,23 @@ def finalize(self):
             self.write_results(fn)
         logging.info("Validating checkpoint and backup files")
         checkpoint_valid = validate_checkpoint_files(
-            self.checkpoint_file, self.backup_file)
+            self.checkpoint_file, self.backup_file, self.name)
         if not checkpoint_valid:
             raise IOError("error writing to checkpoint file")
 
+    def getstate(self):
+        """Strips unserializable attributes of sampler
+        """
+        self_dict = self._sampler.__dict__.copy()
+        # Remove multiprocessing / mpi pool-related/ rstate attributes
+        # as they cannot be pickled
+        del self_dict['pool']
+        del self_dict['M']
+        del self_dict['rstate']
```

Comment:
This won't preserve the random seed for when you start back up from a resume. I think wee need to (1) set the random state explicitly. It might pickle on its own, which would be nice, but if it doens't, you'll have to explicitly preserve it and restore it on resume.